pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -268,23 +351,107 @@ def _extract_submatrix(self, variables, constraints):
                     new_col.append(old_new_var_indices[c])
                     new_data.append(e)
             return sp.sparse.coo_matrix(
-                    (new_data, (new_row, new_col)),
-                    shape=(M, N),
-                    )
+                (new_data, (new_row, new_col)),
+                shape=(M, N),
+            )
+
+    def _extract_subgraph(self, variables, constraints):
+        if self.incidence_graph is None:
+            return get_bipartite_incidence_graph(
+                # Does include_fixed matter here if I'm providing the variables?
```

Comment:
Comment for yourself?

Reply:
Comment for myself. Note that this does change behavior. However, I feel that if variables are explicitly provided, they should be part of the graph regardless of whether they are fixed.

Reply:
To clarify: removing the `include_fixed=True` argument changes behavior (to be more rational, in my opinion).